modify-publication/src/main/java/no/unit/nva/publication/modify/ModifyPublicationHandler.java

```diff
@@ -5,108 +5,71 @@
 
     public static final String PUBLICATION_CONTEXT_JSON = "publicationContext.json";
 
-    private final transient PublicationService publicationService;
+    private final PublicationService publicationService;
+    private final ObjectMapper objectMapper = ObjectMapperConfig.objectMapper;
 
     /**
      * Default constructor for MainHandler.
      */
-    @JacocoGenerated
     public ModifyPublicationHandler() {
-        this(PublicationHandler.createObjectMapper(),
-                new RestPublicationService(
+        this(new RestPublicationService(
                         HttpClient.newHttpClient(),
+                    ObjectMapperConfig.objectMapper,
                         new Environment()),
                 new Environment());
     }
 
     /**
      * Constructor for MainHandler.
      *
-     * @param objectMapper objectMapper
      * @param publicationService    publicationService
      * @param environment  environment
      */
-    public ModifyPublicationHandler(ObjectMapper objectMapper, PublicationService publicationService,
+    public ModifyPublicationHandler(PublicationService publicationService,
                                     Environment environment) {
-        super(objectMapper, environment);
+        super(Publication.class, environment);
         this.publicationService = publicationService;
     }
 
     @Override
-    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
-        String authorization;
-        UUID identifier;
-        Publication publication;
-        try {
-            JsonNode event = objectMapper.readTree(input);
-            authorization = Optional.ofNullable(event.at(HEADERS_AUTHORIZATION).textValue())
-                    .orElseThrow(() -> new IllegalArgumentException(MISSING_AUTHORIZATION_IN_HEADERS));
-            identifier = UUID.fromString(Optional.ofNullable(event.at(PATH_PARAMETERS_IDENTIFIER).textValue())
-                    .orElseThrow(() -> new IllegalArgumentException(MISSING_IDENTIFIER_IN_PATH_PARAMETERS)));
-            publication = objectMapper.readValue(event.at(BODY).textValue(), Publication.class);
-        } catch (Exception e) {
-            logError(e);
-            writeErrorResponse(output, BAD_REQUEST, e);
-            return;
-        }
+    protected JsonNode processInput(Publication input, RequestInfo requestInfo, Context context)
+            throws ApiGatewayException {
+        Publication publication = publicationService.updatePublication(
+                RequestUtil.getIdentifier(requestInfo),
+                input,
+                RequestUtil.getAuthorization(requestInfo));
 
-        log("Identifier in path parameters " + identifier.toString());
-        log("Publication in request body " + objectMapper.writeValueAsString(publication));
+        JsonNode publicationJson = objectMapper.valueToTree(publication);
+        addContext(publicationJson);
```

Comment:
Refactor these two commands in one method. Better readability and we do not have methods with side-effects in the processing logic

Reply:
Isn't this the same logic as the FetchPublicationHandler? Maybe abstract and re-use?